### PR TITLE
 Extend workload cluster test to include app-operator per cluster

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#14163

### CODE REVIEW

File: helm/app-operator/templates/rbac.yaml [hunk 1]
```diff
@@ -100,6 +100,13 @@ rules:
     - create
     - delete
     - get
+- apiGroups:
+    -  scheduling.k8s.io/v1
+  resources:
+    - priorityclasses
+  verbs:
+    - create
+    - get
```
Comment (rossf7) [inline helm/app-operator/templates/rbac.yaml:109]: This is needed so app-operator can bootstrap chart-operator.

File: integration/key/key.go [hunk 2]
```diff
@@ -40,7 +44,7 @@ func DefaultCatalogStorageURL() string {
 	return "https://giantswarm.github.io/default-catalog"
 }
 
-func Namespace() string {
+func GiantSwarmNamespace() string {
```
Comment (rossf7) [inline integration/key/key.go:47]: This makes the diff a mess but I think its needed to make it clearer which namespace this is.

File: integration/setup/setup.go [hunk 3]
```diff
@@ -52,18 +52,18 @@ func installResources(ctx context.Context, config Config) error {
...
 		}
 	}
 
-	// Create CRDs. The Chart CRD is created by the operator
 	// for the kubeconfig test that bootstraps chart-operator.
 	crds := []string{
 		"App",
 		"AppCatalog",
 		"AppCatalogEntry",
+		"Chart",
```
Comment (rossf7) [inline integration/setup/setup.go:66]: I couldn't find a way for the test to create the chart CRD. But I think we can live with that.

File: integration/test/app/workload/workload_cluster_test.go [hunk 4]
```diff
@@ -72,19 +80,32 @@ func TestWorkloadCluster(t *testing.T) {
...
-				WaitForDeploy: true,
+				// Install app-operator in the workload cluster namespace.
+				AppCRNamespace: key.WorkloadClusterNamespace(),
+				CatalogName:    key.ControlPlaneTestCatalogName(),
+				Name:           project.Name(),
+				Namespace:      key.WorkloadClusterNamespace(),
+				ValuesYAML:     templates.AppOperatorValues,
+				SHA:            env.CircleSHA(),
+				WaitForDeploy:  true,
+			},
```
Comment (rossf7) [inline integration/test/app/workload/workload_cluster_test.go:97]: This is the new app CR.

File: pkg/project/project.go [hunk 5]
```diff
@@ -33,9 +33,3 @@ func Source() string {
 func Version() string {
 	return version
 }
-
-// WorkloadClusterAppVersion is always 1.0.0 for workload cluster app CRs using Helm 2.
-// For app CRs using Helm 3 we use project.Version().
-func WorkloadClusterAppVersion() string {
-	return "1.0.0"
-}
```
Comment (rossf7) [inline pkg/project/project.go:41]: Dropping dead code. We need this in app-exporter but not here.

**Comment by tomahawk28** [comment]
app-operator in cluster namespace (`app-operator-594c87cc9d-6dk78`) had some issues with kubeconfig as below. 
```
[{"file":"/go/pkg/mod/github.com/giantswarm/kubeconfig/v4@v4.0.0/kubeconfig.go","line":118},{"file":"/go/pkg/mod/github.com/giantswarm/kubeconfig/v4@v4.0.0/kubeconfig.go","line":45},{"file":"/root/project/service/watcher/chartstatus/g8sclient.go","line":44}]},"time":"2021-02-09T17:43:25.229961+00:00"}
2021-02-09T17:43:25.240246713Z stdout F {"caller":"github.com/giantswarm/app-operator/v3/service/watcher/chartstatus/chartstatus.go:77","level":"error","message":"failed to get g8sclient","stack":{"annotation":"Secret `test-workload-kubeconfig` in Namespace `test-workload`","kind":"missingKubeConfigError","stack":[{"file":"/go/pkg/mod/github.com/giantswarm/kubeconfig/v4@v4.0.0/kubeconfig.go","line":118},{"file":"/go/pkg/mod/github.com/giantswarm/kubeconfig/v4@v4.0.0/kubeconfig.go","line":45},{"file":"/root/project/service/watcher/chartstatus/g8sclient.go","line":44}]},"time":"2021-02-09T17:43:25.240052+00:00"}
```
This passed integration tests but can you maybe check this?

File: integration/test/app/workload/workload_cluster_test.go [hunk 6]
```diff
@@ -61,6 +71,21 @@ func TestWorkloadCluster(t *testing.T) {
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-kubeconfig", key.WorkloadClusterNamespace()),
+				Namespace: key.WorkloadClusterNamespace(),
+			},
+		}
+		_, err = config.K8sClients.K8sClient().CoreV1().Secrets(key.WorkloadClusterNamespace()).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("expected nil got %#v", err)
+		}
```
Comment (rossf7) [inline integration/test/app/workload/workload_cluster_test.go:87]: @tomahawk28 Good catch on the errors in the logs. 

I extended the test to create this kubeconfig secret and now the chart CR watcher works for both app-operators. PTAL

**Review by tomahawk28 — APPROVED** [review]
LGTM!

**Pull request merged**